### ISSUE TITLE
Polish workflow for ML classification tasks (SGD classifier, Tensorflow)

### ISSUE THREAD

**Issue by bfla** [buccaneerai/bottlenose#21]
The SGD classifier is already built out on [the rxsgd  #branch](https://github.com/buccaneerai/bottlenose/tree/rxsgd) but the API needs some work before it gets released.

It would also be wise to revisit sklearn and caret to see if Bottlenose should copy their APIs or try to improve on them.

Personally, I think it would be cool if the models were a bit more like a data frame (with some awareness of which features and label columns they were trained on) so that you could **pass data in as JSON objects and the models would be smart enough to infer and remember the shape** of the objects and format them correctly.  This is a bit of a pain point in sklearn and other libraries where they are overly dependent on arrays and the order is very important.

Another pain point is that sklearn **does not maintain the primary key of instances**.  A better implementation would keep indexes intact.

It's also important to make it extremely easy to **measure the performance of the model**.  Ideally, it would be nice to estimate the performance in real-time or as frequently as possible without over consuming CPU.  With SGD, it might be possible to simply track the error or perhaps the Mean Squared Error as the model is trained on more items.  Maybe there could be a way to just run a secondary process that runs over the test set repeatedly to measure the current performance of the predict method at a reasonable interval.

Checklist:
- [x] Build a working SGD `classifier` (or `train`) operator
- [x] Expose the `predict` method in a good way.
- [ ] Decide what the public API should be for input/output
- [ ] Build an operator that implements the input API
- [ ] Build an operator that implements the output API

What else does it need to be useful in real ML projects?
- [ ] Should convert true/false inputs to 0/1
- [ ] Feature binarizer (one-hot-encoding)
- [ ] Add cross validation?
- [ ] Grid search
- [ ] Multi-classifier
- [ ] Test/train split
- [ ] Performance (f1 score from rxstats?)
- [ ] README
- [ ] docs
